src/dodal/devices/i24/pmac.py

```diff
@@ -141,11 +142,42 @@ async def set(self, value: int, wait=True, timeout=None):
         prog_str = f"&2b{value}r"
         assert isinstance(timeout, SupportsFloat) or (
             timeout is None
-        ), f"ProgramRunner does not support calculating timeout itself, {timeout=}"
+        ), f"ProgramRunner does not support calculating timeout itself, {timeout}"
         await self.signal.set(prog_str, wait=wait)
+        # First wait for signal to go to 1, then wait for the scan to finish.
+        await wait_for_value(
+            self.status,
+            ScanState.RUNNING,
+            timeout=DEFAULT_TIMEOUT,
+        )
         await wait_for_value(self.status, ScanState.DONE, timeout)
```

Comment:
Should: The timeout is now wrong. At it's worst this function as a whole won't timeout until `DEFAULT_TIMEOUT` + `timeout`. You can fix this by putting them in their own `asyncio.wait_for`, see for example the `BartRobot`

Reply:
Added a comment explaining the timeout. In short, the name is misleading but forced by the signature of `set`. The second timeout should be the total time the scan should take so that it doesn't timeout while the motion program is still running. The first one is just a check that the scan actually starts so it doesn't need to be as long.